civis/tests/test_futures.py

```diff
@@ -86,40 +95,31 @@ def test_check_message_when_job_is_running(self, *mocks):
         self.assertFalse(result._check_message(message))
 
     @mock.patch(api_import_str, return_value=civis_api_spec_base)
-    def test_set_api_result_result_succeeded(self, mock_api):
-        poller = mock.Mock()
-        api_result = mock.Mock()
-        api_result.state = 'succeeded'
+    def test_set_api_result_succeeded(self, mock_api):
+        poller = _create_poller_mock("succeeded")
 
         result = CivisFuture(poller, (1, 2))
-        result._set_api_result(api_result)
-        assert poller.call_count == 0
+        assert poller.call_count == 1
```

Comment:
I don't understand why this call count changed?

Reply:
I _think_ it's because now `CivisFuture` has the `self._set_job_exception` callback in `__init__`, and that's what calls the poller. I'll double check.